### PR TITLE
[#2174] bump platform maven artifacts to 2023-03

### PR DESCRIPTION
**Pull request by cdietrich**
[#2174] bump platform maven artifacts to 2023-03

### CODE REVIEW

**Comment by cdietrich** [comment]
looks like this needs fix for
https://github.com/eclipse/xtext/issues/2158
or something the like as the bom does not seems to be used correctly

**Comment by cdietrich** [comment]
unfortunately the log does not seems to have enhough information  on this

**Comment by cdietrich** [comment]
checking locally the maven plugin tests indeed seem to take the dev bom from sonatype snapshots and not the local build one

**Comment by LorenzoBettini** [comment]
@cdietrich on my branch https://github.com/eclipse/xtext/tree/lb_2158_cleanup-maven-plugins-ITs I've cleaned up the Xtend maven plugin ITs. I now create a new branch merging my branch with yours and see whether together the tests pass.

However, if you bump the versions in the BOM, this means that the minimal version of Eclipse deps is 2023-03 while the minimal target platform is 2022-03, is that right?

**Comment by cdietrich** [comment]
No In tycho we stay with old version minimal as we did this diverging the last 7 years

**Comment by LorenzoBettini** [comment]
> No In tycho we stay with old version minimal as we did this diverging the last 7 years

What looks strange to me is that the minimal version in the BOM is higher than the minimum target platform. I seemed to recall it was the other way around or that the minimum target platform corresponded to the minimal version in the BOM.

**Comment by cdietrich** [comment]
Minimal target was oxygen until recently although it has no java 11 support but we had

**Comment by LorenzoBettini** [comment]
@cdietrich merging the two branches https://github.com/eclipse/xtext/tree/lb_2158_cd_2174 and running Xtend Maven ITs work on my machine. I pushed the branch, let's see if also the CI is happy

File: org.eclipse.xtext/src/org/eclipse/xtext/generator/trace/node/TracedAccessorsProcessor.java [hunk 1]
```diff
@@ -63,6 +63,7 @@ public void doTransform(MutableClassDeclaration annotatedClass, @Extension Trans
 					if (TracedAccessorsProcessor.TYPES_WITH_GOOD_TO_STRING
 							.contains(rt.getType().getSimpleName().toLowerCase())) {
 						annotatedClass.addMethod(tracerName(getter), it -> {
+							context.setPrimarySourceElement(it, context.getPrimarySourceElement(annotatedClass));
```
Comment (szarnekow) [inline org.eclipse.xtext/src/org/eclipse/xtext/generator/trace/node/TracedAccessorsProcessor.java:66]: Is this intentionally part of this merge request?

Comment (cdietrich) [reply org.eclipse.xtext/src/org/eclipse/xtext/generator/trace/node/TracedAccessorsProcessor.java:66]: No

**Review by LorenzoBettini — APPROVED** [review]

**Comment by LorenzoBettini** [comment]
@cdietrich I see you rebased this branch, so I'm removing the temporary experiment https://github.com/eclipse/xtext/tree/lb_2158_cd_2174

**Pull request merged**